src/recipes/tests/test_recipe_create.py

```diff
@@ -0,0 +1,63 @@
+from django.test import TestCase
+from assertpy import assert_that
+from rest_framework import status
+from rest_framework.test import APIClient
+from .factories import RecipeFactory, IngredientFactory
+from ..models import Recipe
+from ..serializers import RecipeSerializer
+import factory
+
+LIST_SIZE = 3
+
+
+class RecipeCreateTestCase(TestCase):
+    def setUp(self) -> None:
```

Comment:
You could actually do this in setUpTestData
https://docs.djangoproject.com/en/3.1/topics/testing/tools/#django.test.TestCase.setUpTestData

Reply:
Got it, thanks. `setUp` == `beforeEach` and `setUpTestData` == `beforeAll`

Django seems to have it's own "way" for everything. ;-)

Reply:
Turns out I also had to rename my client variable. It seems Django sets the class client _after_ `setUpTestCase` is called so I was getting weird errors on `PUT` calls because the tests were suddenly using `django.test.Client` instead of `rest_framework.test.APIClient`

Reply:
Have a look at https://www.notion.so/travelperk/Python-testing-best-practices-e695bb4167bd4bd585aaf71bbfad35ae#7dbe639fdad74f00b7b4032eb619e0ab

Apart from being a quite detailed document about testing, this paragraph explains the difference between setUp, setUpClass, and setUpTestData (basically missing setUpClass => before each class.. and setUpTestData _a priori_ shouldn't access the DB)